providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/model/FeatureFlag.java

```diff
@@ -23,18 +23,21 @@ public class FeatureFlag {
     private final String defaultVariant;
     private final Map<String, Object> variants;
     private final String targeting;
+    private final Map<String, Object> metadata;
 
     /** Construct a flagd feature flag. */
     @JsonCreator
-    public FeatureFlag(
-            @JsonProperty("state") String state,
-            @JsonProperty("defaultVariant") String defaultVariant,
-            @JsonProperty("variants") Map<String, Object> variants,
-            @JsonProperty("targeting") @JsonDeserialize(using = StringSerializer.class) String targeting) {
+    public FeatureFlag(@JsonProperty("state") String state,
+                       @JsonProperty("defaultVariant") String defaultVariant,
+                       @JsonProperty("variants") Map<String, Object> variants,
+                       @JsonProperty("targeting") @JsonDeserialize(using = StringSerializer.class) String targeting,
+                       @JsonProperty("metadata") Map<String, Object> metadata
```

Comment:
[question] Currently, we do need to provide an empty map everywhere to be backward compatible. Could we also omit this parameter and make it optional? more a question out of curiosity ;)